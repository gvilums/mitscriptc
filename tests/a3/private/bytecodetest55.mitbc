function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		}
	],
	constants = [34, 30, "foo", 20],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [x, y, h, foo, bar, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	5
		load_func	1
		alloc_closure	0
		store_global	6
		load_func	2
		alloc_closure	0
		store_global	7
		alloc_record
		dup
		dup
		load_const	0
		field_store	1
		alloc_record
		dup
		load_const	1
		field_store	3
		field_store	2
		store_global	0
		load_global	0
		load_const	2
		field_store	4
		load_global	0
		field_load	2
		load_global	0
		field_load	4
		load_const	3
		index_store
		load_global	5
		load_global	0
		call	1
		pop
	]
}