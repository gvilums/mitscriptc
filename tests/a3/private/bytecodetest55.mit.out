{bar:foo h:{foo:20 } y:34 }
